### ISSUE TITLE
Authorization fails with wrong error message

### ISSUE THREAD

**Issue by harunsmrkovic** [digime/digime-sdk-ios#160]
## Checklist
- [x] I’m using the latest version of the digi.me Private Sharing SDK.
- [x] I'm using the latest version of the digi.me App.
- [x] I've searched for existing [GitHub issues](https://github.com/digime/digime-sdk-ios/issues)
- [x] I've looked for existing answers in the [SDK Documentation](https://digime.github.io/digime-sdk-ios/index.html) and [Developer FAQs](https://digime.freshdesk.com/support/solutions/9000115894) 
- [x] This issue does not relate to security and can be disclosed publicly on Github. 
## Environment
Describe your development environment here. Please provide as much information as you can, noting the following details if possible:
- Xcode Version: `X.X.X`
- Swift Version: `X.X.X` (if applicable)
- Installation Platform & Version: `[Cocoapods]` version `X.X.X`
## Goals
What are you trying to achieve? 
Authorize
## Expected Results
What do you expect to happen?
Authorized
## Actual Results
What actually happened?
Not authorized, and also getting a wrong error message
## Steps to Reproduce
What are the steps necessary to reproduce this issue?
Ask for authorization via 3rd party app, but issue is not always reproducible, so not sure
## Code Samples & Details
If possible, please provide a code sample and any additional details to help us identify this issue. 

Log

```
(Console Logs) 26-12 12:31:16.194 Instabug - REACT LOG: 2019-12-26 12:31:16.194 [info][tid:com.facebook.react.JavaScript] '[digi.me] Authorization failed', [Error: User cancelled authorization.]
(Console Logs) 26-12 12:31:16.194 Instabug - REACT LOG: 2019-12-26 12:31:16.193 [warn][tid:main][NativeBridge.m:65] [digime] **Authorization failed User cancelled authorization**.
(Console Logs) 26-12 12:30:55.200 Instabug - REACT LOG: 2019-12-26 12:30:55.200 [info][tid:com.facebook.react.NativeBridgeQueue][NativeBridge.m:59] [digime] Authorize inited
(Console Logs) 26-12 12:30:55.199 Instabug - REACT LOG: 2019-12-26 12:30:55.199 [info][tid:com.facebook.react.JavaScript] [digi.me] Initiated authorization via SDK
```

Important thing to note about the error above is that the user **did not** cancel authorization.

There was also an error received from getAccounts call around the time of this authorization request, it might be related

```
(Console Logs) 26-12 12:28:56.664 Instabug - REACT LOG: 2019-12-26 12:28:56.664 [info][tid:com.facebook.react.JavaScript] '[digi.me] Get Accounts failed', [Error: The operation couldn’t be completed. (me.digi.sdk.api error 404.)]
```

After user authorized, session ID was `4rex9v89Nb7oAkw59iGOISsKRmp96EE2 `

**Comment by harunsmrkovic** [comment]
One more user report, this one is saying that he cannot install (button in our app is showing he has no app)

```
(Console Logs) 29-12 23:51:21.848 Instabug - REACT LOG: 2019-12-29 17:51:21.834 [info][tid:com.facebook.react.JavaScript] '[digi.me] Authorization failed', [Error: cancelled]
(Console Logs) 29-12 23:51:21.834 Instabug - REACT LOG: 2019-12-29 17:51:21.831 [warn][tid:main][NativeBridge.m:65] [digime] Authorization failed cancelled
(Console Logs) 29-12 23:51:21.340 Instabug - REACT LOG: 2019-12-29 17:51:21.339 [info][tid:com.facebook.react.NativeBridgeQueue][NativeBridge.m:59] [digime] Authorize inited
(Console Logs) 29-12 23:51:21.334 Instabug - REACT LOG: 2019-12-29 17:51:21.323 [info][tid:com.facebook.react.JavaScript] [digi.me] Initiated authorization via SDK
(Console Logs) 29-12 23:51:03.275 Instabug - REACT LOG: 2019-12-29 17:51:03.275 [info][tid:com.facebook.react.JavaScript] '[digi.me] Authorization failed', [Error: cancelled]
(Console Logs) 29-12 23:51:03.270 Instabug - REACT LOG: 2019-12-29 17:51:03.267 [warn][tid:main][NativeBridge.m:65] [digime] Authorization failed cancelled
(Console Logs) 29-12 23:51:02.821 Instabug - REACT LOG: 2019-12-29 17:51:02.814 [info][tid:com.facebook.react.NativeBridgeQueue][NativeBridge.m:59] [digime] Authorize inited
(Console Logs) 29-12 23:51:02.813 Instabug - REACT LOG: 2019-12-29 17:51:02.813 [info][tid:com.facebook.react.JavaScript] [digi.me] Initiated authorization via SDK
```

**Comment by brokenzz** [comment]
Found a case where the wrong result is being sent back to the SDK (`CANCEL` instead of `ERROR`). We will release a fix in digi.me app. No SDK changes required for this one.

**Comment by brokenzz** [comment]
Fixed in `4.3.0` Release.
